src/cmd/flux-update.py

```diff
@@ -0,0 +1,225 @@
+##############################################################
+# Copyright 2023 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+import argparse
+import json
+import logging
+import math
+import sys
+
+import flux
+import flux.job
+import flux.util
+
+LOGGER = logging.getLogger("flux-update")
+
+
+class JobspecUpdates:
+    """
+    Convenience class for building a jobspec-update payload from a
+    set of KEY=VALUE pairs on the command line, and a method to send
+    the update as a request to the job manager.
+    """
+
+    #  Mapping of short key names, i.e. as given on the command line,
+    #  to full dotted-path location in jobspec.
+    #
+    #  Note: If a key doesn't exist in this mapping, but also does not start
+    #  with 'attributes.', 'resources.' or 'tasks.', then 'attributes.system'
+    #  is assumed.
+    #
+    key_aliases = {"name": "attributes.system.job.name"}
+
+    def __init__(self, jobid, flux_handle=None):
+        self._flux_handle = flux_handle
+        self.jobid = jobid
+        self.updates = None
+        self.jobspec = None
+
+    @property
+    def flux_handle(self):
+        if self._flux_handle is None:
+            self._flux_handle = flux.Flux()
+        return self._flux_handle
+
+    def _apply_jobspec_updates(self, eventlog):
+        """
+        Apply jobspec updates from eventlog to internal jobspec:
+        """
+        for entry in eventlog.splitlines():
+            event = flux.job.EventLogEvent(entry)
+            if event.name == "jobspec-update":
+                for key, value in event.context.items():
+                    self.jobspec.setattr(key, value)
+
+    def _fetch_jobspec(self, key):
+        """
+        Fetch dotted key 'key' in jobspec for this job, fetching jobspec
+        and eventlog (to apply jobspec-updates) if necessary.
+        """
+        if self.jobspec is None:
+            lookup = flux.job.job_kvs_lookup(
+                self.flux_handle, jobid=self.jobid, keys=["jobspec", "eventlog"]
+            )
+            self.jobspec = flux.job.JobspecV1(**lookup["jobspec"])
+            self._apply_jobspec_updates(lookup["eventlog"])
+
+        return self.jobspec.getattr(key)
```

Comment:
ugh, this seems annoying.  Perhaps we need a way to get the redacted jobspec?  Since `job-list` is already building its own version of this, perhaps some mechanism through there?  Perhaps could write up an issue and can be a TODO item later.

Reply:
Yeah, at some point someone is going to use `flux job info JOBID jobspec` and get the wrong information. But this is what we have for now.